pkg/cmd/codespace/ssh.go

```diff
@@ -298,6 +306,118 @@ func checkAndUpdateOldKeyPair(sshContext ssh.Context) *ssh.KeyPair {
 	return nil
 }
 
+// hasUploadedValidCustomKey checks which private keys are used in ssh configration and
+// compares them to uploaded public keys to see if there is a match
+func hasUploadedPublicKeyForConfig(
+	ctx context.Context,
+	sshContext ssh.Context,
+	apiClient apiClient,
+	customConfigFile string,
+	customHost string,
+) bool {
+	configuredPrivateKeyPaths, err := getConfiguredPrivateKeys(ctx, customConfigFile, customHost)
+	if err != nil {
+		return false
+	}
+
+	configuredPublicKeys := []string{}
+	for _, privateKeyPath := range configuredPrivateKeyPaths {
+		publicKeyPath := privateKeyPath + ".pub"
+
+		publicKeyContent, err := os.ReadFile(publicKeyPath)
+		if err != nil {
+			// The default configuration includes standard keys like id_rsa,
+			// id_ed25519, etc, but these may not actually exist so just skip
+			continue
+		}
+
+		configuredPublicKeys = append(configuredPublicKeys, string(publicKeyContent))
+	}
+
+	if len(configuredPublicKeys) == 0 {
+		// There are no local private keys which ssh would use
+		return false
+	}
+
+	user, err := apiClient.GetUser(ctx)
+	if err != nil {
+		return false
```

Comment:
If this fails (network blip, api issue, etc), we'll likely see a problem with the user being able to SSH somewhere downstream of this process, is it worth swallowing the error? Can't we return it? It may make it harder to figure out user reports otherwise.

Reply:
Added it - we could *almost* get away with removing it entirely because there is just a `/user/keys` API method that doesn't need the username, but it requires a non-default token scope. Oddly, `/users/<username>/keys` does NOT require that scope... 🤔